internal/cluster/cluster.go

```diff
@@ -777,10 +778,30 @@ func (m *Manager) DeleteClusterFiles(ctx context.Context, c *common.Cluster, obj
 	return nil
 }
 
-func (m Manager) PermanentClustersDeletion(ctx context.Context, olderThen strfmt.DateTime, objectHandler s3wrapper.API) error {
+func (m Manager) DeregisterInactiveCluster(ctx context.Context, inactiveSince strfmt.DateTime) error {
+	log := logutil.FromContext(ctx, m.log)
+
+	var clusters []*common.Cluster
+	db := m.db.Unscoped()
+	if reply := db.Where("updated_at < ?", inactiveSince).Find(&clusters); reply.Error != nil {
+		return reply.Error
+	}
+	for i := range clusters {
```

Comment:
```suggestion
	for _, c := range clusters {
```
then you will not need `c := clusters[i]`